tests/rptest/services/redpanda.py

```diff
@@ -439,17 +445,15 @@ def to_cli(self, *, dedicated_node):
         else:
             memory_mb = self._memory_mb
 
-        if self._bypass_fsync is None and not dedicated_node:
-            bypass_fsync = True
-        else:
-            bypass_fsync = self._bypass_fsync
+        bypass_fsync = True if dedicated_node else self._bypass_fsync
```

Comment:
No functional change here just  a simplification: `_bypass_fsync` is a pure `bool`, never `None`.

Reply:
really this should then be:

`bypass_fsync = self._bypass_fsync` because the if branch never can be true?

Reply:
Why can `dedicated_node` never be true? Isn't the right thing to invert the if statement?

```
bypass_fsync = True if not dedicated_node else self._bypass_fsync
```

Reply:
`dedicated_node` can be true but `self._bypass_fsync is None` as per Travis' comment and logic further above can never be true and hence the whole statement?

Reply:
Ah I see yeah.

Reply:
So after your fix we are always effectively running with fsync, right, same as before my change right? Sounds good though It hink the intent is to run with fsync off on non-dedicated by default.

Reply:
Yeah I checked one docker DT run from before your change and we did run with fsync. Originally I did also expect that not to be the case.

Reply:
Here's sort of where the bulk of logic was introduced: https://github.com/redpanda-data/redpanda/pull/4376/files#diff-3f567de33923dec6fa4ca0462aa968416b58c956c824dd007cff23498c342785

I actually don't understand the `None` handling at all: for one, in `to_cli` `self._bypass_fsync` will _never_ be `None` right? So the whole `self._bypass_fsync is None and not dedicated_node` line will never take effect. Even if assume that the stuff in __init__ which makes it non-None was an oversight, then this just sets bypass fsync to false on dedicated nodes, but that would be the default anyway. Perhaps it's a copy paste from the above lines, not sure.

Reply:
Yeah not sure, certainly does look a bit copy and pasty. But in any case it seems before we dedicated/cloud nodes were introduced there was no intention of disabling fsync in docker so everything seems as expected?